src/gsy_e/models/area/scm_manager.py

```diff
@@ -0,0 +1,248 @@
+        return (
+            self.allocated_community_energy_kWh - self.energy_need_kWh
+            if self.allocated_community_energy_kWh > self.energy_need_kWh
+            else 0.0)
+
+    def create_buy_trade(self, current_time_slot: DateTime, seller_name: str,
+                         traded_energy_kWh: float, trade_price_cents: float) -> None:
+        """Create and save a trade object for buying energy."""
+        trade = Trade(
+            str(uuid4()), current_time_slot, None,
+            seller_name, self.home_name,
+            traded_energy=traded_energy_kWh, trade_price=trade_price_cents, residual=None,
+            offer_bid_trade_info=None,
+            seller_origin=seller_name,
+            buyer_origin=self.home_name, fee_price=0., buyer_origin_id=self.home_uuid,
+            seller_origin_id=None, seller_id=None, buyer_id=self.home_uuid,
+            time_slot=current_time_slot)
+        self.trades.append(trade)
+        logging.info("[SCM][TRADE][BID] [%s] [%s] %s", self.home_name, trade.time_slot, trade)
+
+    def create_sell_trade(self, current_time_slot: DateTime, buyer_name: str,
+                          traded_energy_kWh: float, trade_price_cents: float):
+        """Create and save a trade object for selling energy."""
+        trade = Trade(
+            str(uuid4()), current_time_slot, None,
+            self.home_name, buyer_name,
+            traded_energy=traded_energy_kWh, trade_price=trade_price_cents, residual=None,
+            offer_bid_trade_info=None,
+            seller_origin=self.home_name,
+            buyer_origin=buyer_name, fee_price=0., buyer_origin_id=None,
+            seller_origin_id=self.home_uuid, seller_id=self.home_uuid, buyer_id=None,
+            time_slot=current_time_slot)
+        self.trades.append(trade)
+        logging.info("[SCM][TRADE][OFFER] [%s] [%s] %s", self.home_name, trade.time_slot, trade)
+
+
+@dataclass
+class CommunityData:
+    # pylint: disable=too-many-instance-attributes
+    """Represent the energy related statistics of a community."""
+    community_uuid: str
+    consumption_kWh: float = 0.
+    production_kWh: float = 0.
+    self_consumed_energy_kWh: float = 0.
+    energy_surplus_kWh: float = 0.
+    energy_need_kWh: float = 0.
+    energy_bought_from_community_kWh: float = 0.
+    energy_sold_to_grid_kWh: float = 0.
+
+
+@dataclass
+class HomeEnergyBills:
+    """Represents home energy bills."""
+    base_energy_bill: float
+    gsy_energy_bill: float
+    savings_euro: float
+    savings_percent: float
+
+
+class SCMManager:
+    """Handle the community manager coefficient trade."""
+    def __init__(self, community_uuid: str, time_slot: DateTime):
+        self._home_data: Dict[str, HomeAfterMeterData] = {}
+        self.community_data = CommunityData(community_uuid)
+        self._time_slot = time_slot
+        self._bills: Dict[str, HomeEnergyBills] = {}
+        self._grid_fees_reduction = 0.28
```

Comment:
So I understand that this parameter controls the reduction in grid fees as a benefit for the homes to trade on the community and have less interaction with the grid right?
Who came up with this percentage? I guess it will be the same for all homes right? Or will each Home have a separate contract with the DSO? If the first is true, I suggest to add this value in a central entity (e.g. ConstSettings). If the latter is true, we probably need to expose this to the CoefficientArea class, right?

Reply:
> So I understand that this parameter controls the reduction in grid fees as a benefit for the homes to trade on the community and have less interaction with the grid right?

More or less yes, it is a way the DSOs use to incentivise loads to participate in the SCM, therefore the benefit for the loads is not a decrease in energy price like in a "normal" peer-to-peer market, but in a decrease in grid fees. 

> Who came up with this percentage?
This comes from Austrian regulation as far as I know

 > I guess it will be the same for all homes right?
It is the same for all homes, and the same for all SCMs in Austria

> If the first is true, I suggest to add this value in a central entity (e.g. ConstSettings).
I agree and I will adapt it now, thanks!